src/fondant/pipeline/runner.py

```diff
@@ -33,6 +36,39 @@ def run(self, input_spec: str, *args, **kwargs):
 
         subprocess.call(cmd)  # nosec
 
+    def run(
+        self,
+        input: t.Union[Pipeline, str],
+        *,
+        output_path: str = "docker-compose.yml",
```

Comment:
Do we need to expose this? I would move this to a temp file or hidden directory.

Reply:
Wouldn't this make it more difficult to run the pipeline again from the same component spec to follow the "compile once, run many" approach (well in this case it would be  "run once, run many")?

Unless this is temporary until we introduce a general compiler that we can start from as proposed here https://github.com/orgs/ml6team/projects/1/views/8?filterQuery=&pane=issue&itemId=44594106

Reply:
I would only hide this in the Runner. Users can still explicitly compile using the `compile` command or `Compiler` class.

Reply:
With expose you mean the file and not the argument ?

Reply:
Both

Reply:
updated